bindings/node/test/clientEncryption.test.js

```diff
@@ -815,4 +828,220 @@ describe('ClientEncryption', function () {
   it('should provide the libmongocrypt version', function () {
     expect(ClientEncryption.libmongocryptVersion).to.be.a('string');
   });
+
+  describe('createEncryptedCollection()', () => {
+    /** @type {InstanceType<ClientEncryption>} */
+    let clientEncryption;
+    const client = new MockClient();
+    let db;
+    const collectionName = 'secure';
+
+    beforeEach(async function () {
+      clientEncryption = new ClientEncryption(client, {
+        keyVaultNamespace: 'client.encryption',
+        kmsProviders: { local: { key: Buffer.alloc(96, 0) } }
+      });
+
+      db = client.db('createEncryptedCollectionDb');
+    });
+
+    afterEach(async () => {
+      sinon.restore();
+    });
+
+    context('validates input', () => {
+      it('throws if options are omitted', async () => {
+        const error = await clientEncryption
+          .createEncryptedCollection(db, collectionName)
+          .catch(error => error);
+        expect(error).to.be.instanceOf(TypeError);
+      });
+
+      it('throws if options.createCollectionOptions are omitted', async () => {
+        const error = await clientEncryption
```

Comment:
Let's try and make this more specific, maybe looking for a keyword in the error message?

This works now because there are no other nested destructured parameters in the options object.  However, suppose the logic changes to something like this in the future:

```typescript
{
        provider,
        createDataKeyOptions = {},
		option2: { ...fields },
        createCollectionOptions: {
          encryptedFields: { ...encryptedFields },
          ...createCollectionOptions
        }
      }
```

This test would still throw, but it would be throwing because `option2` was undefined, not `createCollectionOptions`.

A more defensive approach would be to create a complete options object that each test can clone and delete any relevant properties from.  then we're sure that each test is only missing the expected property.

Reply:
added regexps to the throw assertions